### PR TITLE
wifi-subsys: icmp_ping

### LINKED ISSUE #144: Wifi-subsys: Configure ping icmp protocol
Add milliseconds between each ping procedure and  a “ping” session contains count procedures

### PR DESCRIPTION
**Pull request by RocioRojas**
### Contribution description
Add milliseconds between each ping procedure
Add a “ping” session contains count procedures
Delete vTaskDelay

### Testing procedure
look at wifi-subsys/components/protocols/icmp_ping.c
- esp_ping_confing_t struct
look at wifi-subsys/main/default_params.h
- #define PING_INT 5000                                /* Milliseconds between each ping procedure*/
  #define PING_RET 3                                /* A “ping” session contains count procedures*/
  #define PING_TIMEOUT 600000              /* Repeat task when timeout finishes*/

 Fixes #144

### CODE REVIEW

File: wifi-subsys/components/protocols/src/icmp_ping.c [hunk 1]
```diff
@@ -90,35 +90,66 @@ void check_on_ping_end(esp_ping_handle_t hdl, void *args)
...
+                        stringlify(PING_TO),
+                        &url, &url_len);
+    if (err != ESP_OK) {
+        ESP_LOGE(__func__,
+                "error to get url the cause is %s",
+                esp_err_to_name(err));
+
+        return err;
+    }
+        int result = getaddrinfo(&url, NULL, &hint, &res);
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/src/icmp_ping.c:111]: bad identation 
```suggestion
    if (err != ESP_OK) {
        ESP_LOGE(__func__,
                "error to get url the cause is %s",
                esp_err_to_name(err));

        return err;
    }
    int result = getaddrinfo(&url, NULL, &hint, &res);
```

File: wifi-subsys/components/protocols/src/icmp_ping.c [hunk 2]
```diff
@@ -90,35 +90,66 @@ void check_on_ping_end(esp_ping_handle_t hdl, void *args)
...
+    if ((result != 0) || (res == NULL)) {
         is_connected = FAIL_BIT;
         is_configured = FAIL_BIT;
-        ESP_LOGE(__func__,"DNS lookup failed err=%d res=%p", result, res);
+        ESP_LOGE(__func__, "DNS lookup failed err=%d res=%p", result, res);
         return ESP_FAIL;
     } else {
         is_configured = CONNECTED_BIT;
-        printf("Initialising ping");
+        printf("Initializing ping");
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/src/icmp_ping.c:119]: change this por esp_logi please i know is not your change, but seems we not see this printf when check icmp PR

**Review by xkevin190 — APPROVED** [review]

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 3]
```diff
@@ -13,51 +13,110 @@
...
 }
 
 TEST_CASE("get current status", "[network]")
 {
-   int is_connected = get_current_status();
-   if (is_connected == 0) {
-       TEST_FAIL();
-   }
+    nvs_init();
+    esp_err_t err = set_default_credentials();
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/test/test_icmp.c:114]: you don't this remove please.

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 4]
```diff
@@ -13,51 +13,110 @@
...
+     esp_err_t err = set_default_credentials();
+    if (err != ESP_OK) {
+        TEST_FAIL();
+    };
+    manual_ping(icmp_callback);
+}
+TEST_CASE("start ping task", "[network]")
+{
+    nvs_init();
+    esp_err_t err = set_default_credentials();
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/test/test_icmp.c:104]: change `set_default_credentials` for  `set_default_ping_settings`  i think you don't need this or yes?

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 5]
```diff
@@ -13,51 +13,110 @@
...
     nvs_init();
-    set_default_credentials();
-    err = wifi_init();
-
-    if (err != ESP_OK) TEST_FAIL();
+    wifi_init();
+    err = set_default_credentials();
+     if (err != ESP_OK) {
+        TEST_FAIL();
+    };
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/test/test_icmp.c:73]:
```suggestion
     err = set_default_credentials();
     if (err != ESP_OK) {
        TEST_FAIL();
    };
    err = wifi_init();
    if (err != ESP_OK) {
        TEST_FAIL();
    };
   
  
```

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 6]
```diff
@@ -13,51 +13,110 @@
...
 }
-
 TEST_CASE("send manual ping", "[network]")
 {
-   manual_ping(icmp_callback);
-//    if (is_connected == 0) {
-//        TEST_FAIL();
-//    }
+    nvs_init();
+     esp_err_t err = set_default_credentials();
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/test/test_icmp.c:95]: check identation

**Review by xkevin190 — CHANGES_REQUESTED** [review]

**Review by xkevin190 — CHANGES_REQUESTED** [review]

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 7]
```diff
@@ -13,51 +13,103 @@
...
+    esp_err_t err = nvs_set_string(stringlify(PING), stringlify(PING_TO), PING_TO);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(__func__,
+                "Error: setting ping address to the nvs %s",
+                esp_err_to_name(err));
+    }
+    err = nvs_set_uint32(stringlify(PING), stringlify(PING_INT), PING_INT);
+    if (err != ESP_OK)
+    {
```
Comment (luisan00) [inline wifi-subsys/components/protocols/test/test_icmp.c:36]: looks like the format of some structures is wrong

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 8]
```diff
@@ -13,51 +13,103 @@
...
+        ESP_LOGE(__func__,
+                "Error: setting ping interval to the nvs %s",
+                esp_err_to_name(err));
+    }
+    err = nvs_set_uint8(stringlify(PING), stringlify(PING_RET), PING_RET);
+    if (err != ESP_OK)
```
Comment (luisan00) [inline wifi-subsys/components/protocols/test/test_icmp.c:42]: here too

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 9]
```diff
@@ -13,51 +13,103 @@
...
 TEST_CASE("get current status", "[network]")
 {
-   int is_connected = get_current_status();
-   if (is_connected == 0) {
-       TEST_FAIL();
-   }
+    int is_connected = get_current_status();
+    if (is_connected == 0) {
+        TEST_FAIL();
+    }
```
Comment (luisan00) [inline wifi-subsys/components/protocols/test/test_icmp.c:114]: indentation ...

File: wifi-subsys/components/protocols/test/test_icmp.c [hunk 10]
```diff
@@ -13,51 +13,103 @@
...
+    err = nvs_set_uint8(stringlify(PING), stringlify(PING_RET), PING_RET);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(__func__,
+                "Error: setting ping retries to the nvs %s",
+                esp_err_to_name(err));
+    }
+    err = nvs_set_uint32(stringlify(PING), stringlify(PING_TIMEOUT), PING_TIMEOUT);
+    if (err != ESP_OK)
```
Comment (luisan00) [inline wifi-subsys/components/protocols/test/test_icmp.c:49]: looks like you are not using the clang format tool correctly

File: wifi-subsys/components/protocols/src/icmp_ping.c [hunk 11]
```diff
@@ -19,155 +19,193 @@
...
 
+        return err;
+        }
+        err = nvs_get_uint32(
+            stringlify(PING), stringlify(PING_INT),
+            &ping_config.interval_ms); // Milliseconds between each ping procedure
+        if (err != ESP_OK) {
+        ESP_LOGE(__func__, "error to get ping time the cause is %s",
+                esp_err_to_name(err));
+
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/src/icmp_ping.c:139]: identation error

File: wifi-subsys/components/protocols/src/icmp_ping.c [hunk 12]
```diff
@@ -19,155 +19,193 @@
...
         freeaddrinfo(res);
         esp_ping_config_t ping_config = ESP_PING_DEFAULT_CONFIG();
-        ping_config.target_addr = target_addr;   // target IP address
-        ping_config.count = 1;    // ping in infinite mode, esp_ping_stop can stop it
+        ping_config.target_addr = target_addr; // target IP address
+        err = nvs_get_uint8(
+            stringlify(PING), stringlify(PING_RET),
+            &ping_config.count); // ping in infinite mode, esp_ping_stop can stop it
+        if (err != ESP_OK) {
+        ESP_LOGE(__func__, "error to get ping count the cause is %s",
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/src/icmp_ping.c:128]: identation error

File: wifi-subsys/components/protocols/src/icmp_ping.c [hunk 13]
```diff
@@ -19,155 +19,193 @@
...
         cbs.cb_args = NULL;
 
-        esp_err_t err = esp_ping_new_session(&ping_config, &cbs, &ping);
+        err = esp_ping_new_session(&ping_config, &cbs, &ping);
         if (err != ESP_OK) {
-            ESP_LOGE(__func__,
-                    "Error to the init ping session %s",
-                    esp_err_to_name(err));
-            return err;
+        ESP_LOGE(__func__, "Error to the init ping session %s",
```
Comment (xkevin190) [inline wifi-subsys/components/protocols/src/icmp_ping.c:151]: identation error

**Review by xkevin190 — CHANGES_REQUESTED** [review]

**Review by xkevin190 — APPROVED** [review]

**Review by xkevin190 — APPROVED** [review]

**Comment by luisan00** [comment]
Squash and ready to go ahead 👍

**Pull request merged**